source/frontend/src/features/properties/map/project/add/ProductsArrayForm.tsx

```diff
@@ -0,0 +1,132 @@
+import { LinkButton, StyledRemoveLinkButton } from 'components/common/buttons';
+import GenericModal from 'components/common/GenericModal';
+import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
+import { Section } from 'features/mapSideBar/tabs/Section';
+import { FieldArray, FieldArrayRenderProps, FormikProps, useFormikContext } from 'formik';
+import { useProductProvider } from 'hooks/providers/useProductProvider';
+import React, { useRef, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaTrash } from 'react-icons/fa';
+import styled from 'styled-components';
+
+import { ProductForm, ProjectForm } from './models';
+import ProductSubForm from './ProductSubForm';
+
+export interface IProductsArrayFormProps {
+  field: string;
+  formikProps: FormikProps<ProjectForm>;
+}
+
+export const ProductsArrayForm: React.FunctionComponent<IProductsArrayFormProps> = ({
+  field,
+  formikProps,
+}) => {
+  const arrayHelpersRef = useRef<FieldArrayRenderProps | null>(null);
+  const { values } = useFormikContext<ProjectForm>();
+
+  const [showModal, setShowModal] = useState(false);
+  const [showFileModal, setShowFileModal] = useState(false);
+  const [rowToDelete, setRowToDelete] = useState<number | undefined>(undefined);
+
+  const { retrieveProductFile, retrieveProductFileLoading } = useProductProvider();
+
+  const products = values.products || [];
+
+  const handleRemove = async (index: number) => {
```

Comment:
huh, interesting. Think I need to look at the story to understand what is going on here.

Reply:
oh... ok. So this relates to: "Important: you cannot delete this product unless you remove it from associated files first."